+sw_tests/+unit_tests/unittest_spinw_spinwave.m

```diff
@@ -0,0 +1,471 @@
+            testCase.swobj.genmagstr('mode', 'direct', 'k', [0 0 0], ...
+                                     'n', [1 0 0], 'S', [0; 1; 0]);
+        end
+        function setup_tri_model(testCase)
+            % Create a simple triangular lattice model
+            testCase.swobj_tri = spinw;
+            testCase.swobj_tri.genlattice('lat_const', [4 4 6], 'angled', [90 90 120]);
+            testCase.swobj_tri.addatom('r', [0 0 0], 'S', 3/2, 'label', 'MCr3');
+            testCase.swobj_tri.genmagstr('mode', 'helical', 'S', [1; 0; 0], ...
+                                         'n', [0 0 1], 'k', [1/3 1/3 0]);
+            J1 = 1;
+            testCase.swobj_tri.addmatrix('label','J1','value',J1);
+            testCase.swobj_tri.gencoupling;
+            testCase.swobj_tri.addcoupling('mat','J1','bond',1);
+        end
+    end
+    methods
+        function sw = get_expected_sw_qh5(testCase)
+            % Expected output for the chain model for 5 q-points from
+            % [0 0 0] to [1 0 0]
+            expected_hkl = testCase.qh5;
+            expected_Sab = zeros(3, 3, 2, 5);
+            Sab1 = [0.5 0  0.5j; 0 0 0; -0.5j 0 0.5];
+            Sab2 = [0.5 0 -0.5j; 0 0 0;  0.5j 0 0.5];
+            expected_Sab(:, :, 1, 1:4) = repmat(Sab1, 1, 1, 1, 4);
+            expected_Sab(:, :, 2, 5) = Sab1;
+            expected_Sab(:, :, 2, 1:4) = repmat(Sab2, 1, 1, 1, 4);
+            expected_Sab(:, :, 1, 5) = Sab2;
+
+            sw = testCase.default_spinwave;
+            sw.omega = [ 1e-5  2.  4.  2. -1e-5; ...
+                        -1e-5 -2. -4. -2.  1e-5];
+            sw.Sab = expected_Sab;
+            sw.hkl = expected_hkl;
+            sw.hklA = expected_hkl*2/3*pi;
+            sw.obj = copy(testCase.swobj);
+        end
+    end
+    % Put tests with mocks in own block - prevent interference with other
+    % tests
+    methods (Test)
+        function test_noInput(testCase)
+            % Tests that if call spinwave with no input, it calls the help
+            % First mock the help call
+            help_function = sw_tests.utilities.mock_function('swhelp');
+            testCase.swobj.spinwave();
+            testCase.assertEqual(help_function.n_calls, 1);
+            testCase.assertEqual(help_function.arguments, {{'spinw.spinwave'}});
+        end
+        function test_sw_qh5_optmem(testCase)
+            qpts = testCase.qh5;
+            optmem = 3;
+            % Test that calculation is split into optmem chunks - sw_timeit
+            % is called on each chunk plus once at beginning and end of
+            % function. This is a bit fragile, may need to change the
+            % target function, number of calls, or not check at all if
+            % spinwave is refactored
+            sw_timeit_mock = sw_tests.utilities.mock_function('sw_timeit');
+            sw_out = testCase.swobj.spinwave(qpts, 'optmem', optmem);
+            testCase.assertEqual(sw_timeit_mock.n_calls, optmem + 2);
```

Comment:
Yes, I agree this is a bit fragile at the moment, but the easiest/best way to do it right now. In future I think the body of each chunk can be refactored into a separate function/method (maybe in C++) so the loop essentially just calls this at each iteration and then you can mock that instead and check how often it is called.